### PR TITLE
Refactor: controller: functionalize record of synthesized action event

### PR DESCRIPTION
**Pull request by gao-yan**
Previously given a scenario:

- rscA failed to migrate from node1 to node2 with a failed migrate_to on
node1
- rscA failed to stop on node2
- node2 got fenced and rscA recovered on node1
- node2 rejoined

, rscA would be considered being "multiple-active" and an unnecessary
full recovery would be issued.

With this commit, when node2 gets fenced, a successful event for the
pseudo stop of rscA will be synthesized in resource history, so that
rscA won't be considered being active on node2 when node2 rejoins which
will prevent unnecessary stop or recovery of rscA.

### CODE REVIEW

**Comment by gao-yan** [comment]
It's a proof of concept per discussion from:
https://github.com/ClusterLabs/pacemaker/pull/2146#pullrequestreview-911922137

I haven't tested it or created a scheduler regression test yet.

Reviews and suggestions are appreciated.

File: lib/pengine/unpack.c [hunk 1]
```diff
@@ -2802,6 +2802,7 @@ unpack_migrate_to_failure(pe_resource_t *rsc, pe_node_t *node, xmlNode *xml_op,
                      target_stop_id, target_migrate_from_id);
         if (target_node && target_node->details->online) {
             native_add_running(rsc, target_node, data_set, FALSE);
+            pe__set_resource_flags(rsc, pe_rsc_active_by_migrate_to);
```
Comment (gao-yan) [inline lib/pengine/unpack.c:2805]: Of course a more precise way could be recording "target_node" which is the exact node where the resource is impliedly active by the migrate_to failure. But that would likely need pe_resource_s to be expanded even more...

**Comment by gao-yan** [comment]
Hrm, overlooked... it seems that the synthesized event will get cleaned up by this when node2 rejoins:
https://github.com/ClusterLabs/pacemaker/blob/main/daemons/controld/controld_join_dc.c#L592

**Comment by gao-yan** [comment]
And given that resource history will get cleaned up also for the case of remote node upon rejoining:
https://github.com/ClusterLabs/pacemaker/blob/main/daemons/controld/controld_remote_ra.c#L196

, this solution seems not really feasible.

**Comment by kgaillot** [comment]
I'm a little confused -- if rscA is successfully stopped on node1 as part of the recovery, doesn't that clear the last_failure?

The first commit here is worthwhile anyway :)

**Comment by gao-yan** [comment]
> I'm a little confused -- if rscA is successfully stopped on node1 as part of the recovery, doesn't that clear the last_failure?

The successful stop on node1 only overrode the "rscA_last_0".  The migrate_to failure “rscA_last_failure_0” is still existing and affecting.

> The first commit here is worthwhile anyway :)

Indeed :)

**Comment by gao-yan** [comment]
Left only the first commit here.

**Comment by kgaillot** [comment]
Thanks!

**Pull request merged**